src/main/java/com/epam/ta/reportportal/dao/ItemAttributeRepositoryCustom.java

```diff
@@ -26,6 +27,16 @@
  */
 public interface ItemAttributeRepositoryCustom {
 
+	/**
+	 * Retrieves {@link com.epam.ta.reportportal.entity.launch.Launch} and {@link com.epam.ta.reportportal.entity.item.TestItem}
+	 * {@link ItemAttribute#getKey()} by project id and part of the {@link ItemAttribute#getKey()}.
+	 *
+	 * @param projectId {@link Project#getId()}
+	 * @param keyPart   Part of the {@link ItemAttribute#getKey()}
+	 * @return {@link List} of matched attribute keys
+	 */
+	List<String> findKeysByProjectId(Long projectId, String keyPart, boolean isSystem);
```

Comment:
should be more appropriate method name
findSimilarKeysByProjectId
or
findKeysWithTerm

the current method name supposes find all keys by project,